LIPs/LIP-25.md

```diff
@@ -0,0 +1,237 @@
+---
+lip: 25
+title: Extensible Governance Contract
+author: Nico Vergauwen (@kyriediculous) <nico@livepeer.org>
+type: Meta
+status: Draft
+created: 2020-06-09
+requires: n/a
+---
+
+## Abstract
+
+This proposal describes an extensible governance system that can be used to update the parameters and code of the Livepeer protocol smart contracts and establish a basis for a clear upgrade path for the governance system to give stakeholders control over the system.
+
+## Motivation
+
+At the moment, a core developer owned multisig has admin privileges over the Livepeer protocol, giving the core developers the ability to update contract code and parameters.
+
+The goal for an extensible governance contract is to establish the technical foundation that allows for a variety of sophisticated binding voting systems to be implemented in the future.
+
+After deployment, ownership of the Controller contract will be transferred to the extensible governance contract which will still have a core developer owned multisig as it's main and only actor until control can be phased to binding voting systems in later milestones of the governance roadmap. 
+
+From a high level this extensible governance system should fulfill following properties: 
+- Enable on-chain access rights allowing particular actors to execute (particular) code/parameter updates
+- Allow future changes to the governance structure
+- Enable on-chain rules for how code/parameters can be updated, e.g. delayed execution.
+- Facilitate atomic execution of multiple upgrades
+
+
+## Specification
+
+The extensible governance system will be a Smart Contract deployed on the Ethereum blockchain consisting of following components
+- Access Control List
+- Staged Execution
+- Batched Execution
+
+### Access Control List
+
+The Access Control List contains rules that grant or deny access to certain actors to execute code/parameter updates in the Livepeer protocol.
+
+This mechanism grants access based on Ethereum addresses, whether that be an EOA, smart contract or multisig. The access rights can be for either the entire scope of the protocol or modular (cfr. root and users). 
+
+![image](https://user-images.githubusercontent.com/22256858/84168478-bc960580-aa77-11ea-8f0d-12376205ad2e.png)
```

Comment:
Let's add assets used in this document to the `assets` directory so that they'll live in the same repo as the document itself. We can store assets for this document in the path `assets/lip-25/`.

Reply:
fixed in e141884